File: PhonoArk/PhonoArk/Models/ExamResult.cs
using System;

namespace PhonoArk.Models;

public class ExamResult
{
    public int Id { get; set; }
    public DateTime ExamDate { get; set; } = DateTime.UtcNow;
    public int TotalQuestions { get; set; }
    public int CorrectAnswers { get; set; }
    public string ExamScope { get; set; } = "All"; // "All", "Favorites", etc.
    public TimeSpan Duration { get; set; }
    
    public double Score => TotalQuestions > 0 ? (double)CorrectAnswers / TotalQuestions * 100 : 0;
}

File: pureopus/app/src/main/java/com/phonoark/data/model/ExamResult.kt
package com.phonoark.data.model

import java.time.LocalDateTime

data class ExamResult(
    val id: Long = 0,
    val examDate: LocalDateTime = LocalDateTime.now(),
    val totalQuestions: Int,
    val correctAnswers: Int,
    val examScope: String = "All",
    val durationSeconds: Long = 0,
    val attempts: List<ExamQuestionAttempt> = emptyList()
) {
    val scorePercentage: Double
        get() = if (totalQuestions > 0) correctAnswers.toDouble() / totalQuestions * 100 else 0.0
}
